src/hooks/useViewportRows.ts

```diff
@@ -94,20 +94,79 @@ export function useViewportRows<R>({
     }
   }, [expandedGroupIds, groupedRows, rawRows]);
 
+  const { totalRowHeight, getRowTop, getRowHeight, findRowIdx } = useMemo(() => {
+    if (typeof rowHeight === 'number') {
+      return {
+        totalRowHeight: rowHeight * rows.length,
+        getRowTop: (rowIdx: number) => rowIdx * rowHeight,
+        getRowHeight: () => rowHeight,
+        findRowIdx: (offset: number) => Math.floor(offset / rowHeight)
+      };
+    }
+
+    const rowPositions: ({ height: number; top: number })[] = [];
+    let totalRowHeight = 0;
+    // Calcule the height of all the rows upfront. This can cause performance issues
+    // and we can consider using a similar approach as react-window
+    // https://github.com/bvaughn/react-window/blob/master/src/VariableSizeList.js#L68
+    rows.forEach((row: R | GroupRow<R>) => {
+      const currentRowHeight = isGroupRow(row)
+        ? rowHeight({ type: 'GROUP', row })
+        : rowHeight({ type: 'ROW', row });
+      rowPositions.push({ top: totalRowHeight, height: currentRowHeight });
+      totalRowHeight += currentRowHeight;
+    });
+
+    return {
+      totalRowHeight,
+      getRowTop(rowIdx: number) {
+        if (rowIdx < 0) {
+          rowIdx = 0;
+        } else if (rowIdx >= rows.length) {
+          rowIdx = rows.length - 1;
+        }
+        return rowPositions[rowIdx].top;
```

Comment:
Do we ever call `getRowTop` when we have 0 rows?

Reply:
We can from outside https://github.com/adazzle/react-data-grid/blob/5b0125ce6052cc803caf28d3024e732865294c23/src/DataGrid.tsx#L349